src/schedule/schedule.c

```diff
@@ -75,3 +76,38 @@ void scheduler_init(int type, const struct scheduler_ops *ops, void *data)
 
 	scheduler_register(sch);
 }
+
+void scheduler_get_task_info(struct scheduler_props *scheduler_props,
+			     uint32_t *data_off_size,
+			     struct list_item *tasks)
+{
+	/* TODO
+	 * - container_of(tlist, struct task, list)->uid->id could possibly be used as task id,
+	 *   but currently accessing it crashes, so setting the value to 0 for now
+	 */
+	struct task_props *task_props;
+	struct list_item *tlist;
+
+	scheduler_props->core_id = cpu_get_id();
+	scheduler_props->task_count = 0;
+	*data_off_size += sizeof(*scheduler_props);
+
+	task_props = (struct task_props *)((uint8_t *)scheduler_props + sizeof(*scheduler_props));
+	list_for_item(tlist, tasks) {
```

Comment:
do any locks have to be held while traversing the list

Reply:
I only do reading here but if the list size changes in some other thread that might be a problem. I'm guessing i should just use the locks to be safe. Not in this function though, but in the scheduler specific functions that call this general one, as each scheduler uses different locks. I'll get to it.

Reply:
Added locks to zephyr_ll and dp files, where this function is called.